src/checkpoint.hh

```diff
@@ -0,0 +1,432 @@
+     stream << indent() << "if (curr_file.find(tname) != curr_file.npos) {"
+            << endl;
+     inc_indent();
+     stream << indent() << "return entry.path();" << endl;
+     dec_indent();
+     stream << indent() << "}" << endl;
+     dec_indent();
+     stream << indent() << "}" << endl;
+     stream << indent() << "return boost::filesystem::path(\"\");" << endl;
+     dec_indent();
+     stream << indent() << "}" << endl << endl;
+     dec_indent(); dec_indent();
+  }
+
+  void init(Printer::Base &stream) {
+     inc_indent(); inc_indent(); inc_indent();
+     stream << indent() << "std::string executable_name = "
+            << "boost::dll::program_location().filename().string();" << endl;
+     stream << indent() << "int process_id = getpid();" << endl;
+     stream << indent() << "std::string archive_name = executable_name + "
+            << "\"_\" + std::to_string(process_id);" << endl;
+     stream << indent() << "table_path = out_path / (archive_name + \"_\" + "
+            << "tname);" << endl;
+     stream << indent() << "std::string in_path_error_msg = "
+            << "\"at user-specified path\";" << endl << endl;
+     stream << indent() << "if (in_path.empty()) {" << endl;
+     inc_indent();
+     stream << indent() << "in_path = "
+            << "parse_checkpoint_log(out_path, tname, executable_name, "
+            << "arg_string);" << endl;
+     stream << indent() << "in_path_error_msg = \"in Logfile\";" << endl;
+     dec_indent();
+     stream << indent() << "} else {" << endl;
+     inc_indent();
+     stream << indent() << "in_path = find_table(in_path, tname);" << endl;
+     dec_indent();
+     stream << indent() << "}" << endl << endl;
+     stream << indent() << "// read the DP array/table from disk "
+                            "and put its contents into array" << endl;
+     stream << indent() << "try {" << endl;
+     inc_indent();
+     stream << indent() << "std::ifstream array_fin(in_path, "
+                            "std::ios::binary);" << endl;
+     stream << indent() << "if (!(array_fin.good())) {" << endl;
+     stream << indent() << "  throw std::ifstream::failure(\"\");" << endl;
+     stream << indent() << "}" << endl;
+     stream << indent() << "boost::archive::binary_iarchive "
+                            "array_in(array_fin);" << endl;
+     stream << indent() << "array_in >> array;" << endl;
+     stream << indent() << "array_fin.close();" << endl;
+     stream << indent() << "// mark the already existing table values "
+                            "in the tabulated vector" << endl;
+     stream << indent() << "for (long unsigned int i = 0; i < array.size(); "
+                            "i++) tabulated[i] = array[i];" << endl;
+     dec_indent();
+     stream << indent() << "} catch (const std::ifstream::failure &e) {"
+            << endl;
+     stream << indent() << "  std::cerr << \"Info: \\\"\" + tname + \"\\\" "
```

Comment:
if we print this type of message, shouldn't we also create an info message every time a table was archived?

Reply:
Sure, we can do that.